File: oracles/python/examples/read_oracle_snapshot.py
from __future__ import annotations

import os

from chancery_reference.client import ChanceryClient
from oracle_snapshot import build_oracle_snapshot


def required_environment_variable(name: str) -> str:
    value = os.environ.get(name)
    if value is None or value == "":
        raise ValueError(f"Missing environment variable {name}")
    return value


def main() -> int:
    client = ChanceryClient(
        required_environment_variable("RPC_URL"),
        "confirmed",
    )
    discovery = client.discover()
    print(ChanceryClient.stringify(build_oracle_snapshot(discovery)))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: oracles/typescript/examples/ReadOracleSnapshot.ts
import { ChanceryClient } from "../../../typescript/src/index.js";
import { buildOracleSnapshot } from "../src/OracleSnapshot.js";

function requiredEnvironmentVariable(name: string): string {
    const value = process.env[name];
    if (value === undefined || value.length === 0) {
        throw new Error(`Missing environment variable ${name}`);
    }
    return value;
}

async function main(): Promise<void> {
    const client = new ChanceryClient(
        requiredEnvironmentVariable("RPC_URL"),
        "confirmed",
    );
    const discovery = await client.discover();
    process.stdout.write(
        `${ChanceryClient.stringify(buildOracleSnapshot(discovery))}\n`,
    );
}

main().catch((error: unknown) => {
    const message = error instanceof Error ? error.stack ?? error.message : String(error);
    process.stderr.write(`${message}\n`);
    process.exitCode = 1;
});